src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/ViewHelper.java

```diff
@@ -31,10 +32,35 @@ public static void setPivotYIfNeeded(View view, float pivotY)
             view.setPivotY(pivotY);
     }
 
+    public static void setContentDescriptionForAutomationId(View view, String description)
+    {
+        view = getSemanticNativeElement(view);
+
+        // Android automatically sets ImportantForAccessibility to "Yes" when you set the ContentDescription.
+		// We are only setting the ContentDescription here for Automation testing purposes so
+		// we don't want Layouts/images/Other controls to automatically toggle to Yes.
+		// Unfortunately Android (AFAICT) doesn't have an obvious way of calculating what "Auto" will be interpreted as
+		// iOS is kind of enough to indicate that anything inheriting from "UIControl" but the Android documentation
+		// just says "Android uses heuristics to figure out what Auto will mean"
+		// It seems like if we just toggle this back to "Auto" that everything just works.
+        int importantForAccessibility = view.getImportantForAccessibility();
+        view.setContentDescription(description);
+        if (importantForAccessibility == View.IMPORTANT_FOR_ACCESSIBILITY_AUTO) {
```

Comment:
If it's already auto, why do we need to set it to auto? What am I missing here?

Reply:
when you set the `ContentDescription` android automatically flips `ImportantForAccessibility` to `Yes`

If you were to step through this code and inspect `view.getImportantForAccessibility()` on like 48 it would now equal `Yes` it would no longer equal `Auto` so we are resetting it back to `Auto` to undo the side effect of androids changes

Reply:
yes, so shouldn't we be checking here that it's YES instead of AUTO?
```suggestion
        if (importantForAccessibility == View.IMPORTANT_FOR_ACCESSIBILITY_YES) {
```

Reply:
If it's `YES` before getting to this method then why would we revert it back to `AUTO`?  We would just want it to remain as 'YES`. All we're doing here is reverting back to the default ( which is AUTO). If the view coming in hasn't been explicitly set one way or the other than us setting `contentDesc` forces the view away from default. The idea here is that we want to remove any side effects caused by setting the `contentDesce`